internal/repository/user_email_repository.go

```diff
@@ -0,0 +1,112 @@
+	"go.uber.org/zap"
+)
+
+type EmailStore struct {
+	db *pgxpool.Pool
+}
+
+type EmailStorer interface {
+	GetInvitations(ctx context.Context, profileID int, tx pgx.Tx) (specs.InvitationResponse, error)
+	CreateInvitation(ctx context.Context, invitation Invitations, tx pgx.Tx) error
+	UpdateProfileCompleteStatus(ctx context.Context, profileID int, updateReq UpadateRequest, tx pgx.Tx) error
+}
+
+func NewUserEmailRepo(db *pgxpool.Pool) EmailStorer {
+	return &EmailStore{
+		db: db,
+	}
+}
+
+func (userStore *EmailStore) GetInvitations(ctx context.Context, profileID int, tx pgx.Tx) (specs.InvitationResponse, error) {
+	queryBuilder := psql.Select(constants.RequestInvitationColumns...).From("invitations").Where(sq.And{sq.Eq{"profile_id": profileID}, sq.Eq{"is_profile_complete": 0}})
+	sql, args, err := queryBuilder.ToSql()
+	if err != nil {
+		zap.S().Error("Error generating select query for get invitations: ", err)
+		return specs.InvitationResponse{}, err
+	}
+	var Invitation specs.InvitationResponse
+	err = tx.QueryRow(ctx, sql, args...).Scan(
+		&Invitation.ProfileID, &Invitation.ProfileComplete, &Invitation.CreatedAt, &Invitation.UpdatedAt, &Invitation.CreatedByID, &Invitation.UpdatedByID,
+	)
+	if err != nil {
+		if err == pgx.ErrNoRows {
+			zap.S().Info("No invitations found with profile ID %d", profileID)
+			return specs.InvitationResponse{}, err
+		}
+		zap.S().Error("Error executing query for get invitations: ", err)
+		return specs.InvitationResponse{}, err
+	}
+	return Invitation, nil
+}
+
+func (emailStore *EmailStore) CreateInvitation(ctx context.Context, invitation Invitations, tx pgx.Tx) error {
+	values := []interface{}{
+		invitation.ProfileID, invitation.ProfileComplete, invitation.CreatedAt, invitation.UpdatedAt,
+		invitation.CreatedByID, invitation.UpdatedByID,
+	}
+
+	insertQuery, args, err := psql.Insert("invitations").
+		Columns(constants.RequestInvitationColumns...).
+		Values(values...).
+		Suffix("RETURNING id").
+		ToSql()
+	if err != nil {
+		zap.S().Error("Error generating email invitation insert query: ", err)
+		return err
+	}
+
+	if tx == nil {
+		zap.S().Error("Transaction is nil")
+		return fmt.Errorf("internal error: transaction is nil")
+	}
+
+	res, err := tx.Exec(ctx, insertQuery, args...)
+	if err != nil {
+		if helpers.IsDuplicateKeyError(err) || helpers.IsInvalidProfileError(err) {
+			zap.S().Info("Error creating send invitation: ", err)
+			return err
+		}
+		zap.S().Error("Error executing insert query: ", err)
+		return err
+	}
+
+	if res.RowsAffected() == 0 {
+		zap.S().Info("No rows affected with given query : ", insertQuery)
+		return nil
+	}
+
+	return nil
+}
+
+func (emailStore *EmailStore) UpdateProfileCompleteStatus(ctx context.Context, profileID int, updateReq UpadateRequest, tx pgx.Tx) error {
+	queryBuilder := psql.Update("invitations").Set("is_profile_complete", updateReq.ProfileComplete).Set("updated_at", updateReq.UpdatedAt).Where(sq.Eq{"profile_id": profileID, "is_profile_complete": 0})
```

Comment:
same in profileID, "is_profile_complete": 0, pass into filter

Reply:
done